lib/api/src/externals/function.rs

```diff
@@ -67,6 +67,49 @@ pub struct Function {
     pub(crate) exported: ExportFunction,
 }
 
+fn build_export_function_metadata<Env>(
+    env: Env,
+    import_init_function_ptr: for<'a> fn(
+        &'a mut Env,
+        &'a crate::Instance,
+    ) -> Result<(), crate::HostEnvInitError>,
```

Comment:
What is the `for` syntax? First time I'm seeing this!

Reply:
Haha yeah, it's not super common

it's used to bind a lifetime in a scope, it's a universal qualifier (like ∀), it's just saying "for all lifetimes `'a`, this thing, it's a way of constraining the type (and it's required here because it'll be inferred incorrectly)

Reply:
Yes, obviously there is a difference. The difference with `'a` on the function itself is that it will be inferred from the calling context. That's not what we want! Forget it.